### DISCUSSION TITLE
Convert output of one retrieval QA chain as the prompt for a second retrieval QA chain

### DISCUSSION THREAD

**Discussion by rvy474** [FlowiseAI/Flowise#1434] [Q&A] [answered]
I have created a retrieval QA chain 1. This is create by embedding+splitting PDF 1 into a vector DB and using the retrieval QA chain to get relevant answers. 
Id like it so that the output of this QA chain 1 is now used used to find relevant sections in PDF 2. Which is also loaded into a vector DB.

Id like the response to contain tthe outputs of both QA chains if possible.

**Comment by Jaredude** [comment, marked as answer]
You could accomplish this by wrapping everything into a Custom Tool and using that Custom Tool in your Chat agent. You'd take the input from the user and pass that to the Custom Tool. The Custom Tool would then call the first QA chain via the API and use the results of that to then call the second QA chain via the API. The Custom Tool would then return the results of both API calls. Hope that helps.

**Reply by dkindlund** [reply]
@Jaredude , how do you pass the user's entire "conversational history" as a parameter into the Custom Tool?

**Reply by dkindlund** [reply]
Ah, I think I figured it out -- you have to specify the `"{{chat_history}}"` as a valid input into the `Custom Tool`.
